File: 1267-count-servers-that-communicate/1267-count-servers-that-communicate.py
class Solution:
    def countServers(self, grid: List[List[int]]) -> int:
        m, n = len(grid), len(grid[0])
        
        # count servers in each row & col
        rows, cols = [0] * m, [0] * n
        for i in range(m):
            for j in range(n):
                rows[i] += grid[i][j]
                cols[j] += grid[i][j]

        # find out servers that can communicate
        count = 0
        for i in range(m):
            for j in range(n):
                if grid[i][j] == 0 or (rows[i] == 1 and cols[j] == 1): continue
                count += 1

        return count

File: 1267-count-servers-that-communicate/1267-count-servers-that-communicate.cs
public class Solution {
    public int CountServers(int[][] grid) {
        int m = grid.Length, n = grid[0].Length;
        
        // count servers in each row & col
        int[] rows = new int[m], cols = new int[n];
        for(var i=0; i<m; i++){
            for(var j=0; j<n; j++){
                rows[i] += grid[i][j];
                cols[j] += grid[i][j];
            }
        }

        // find out servers that can communicate
        var count = 0;
        for(var i=0; i<m; i++){
            for(var j=0; j<n; j++){
                if(grid[i][j] == 0 || (rows[i] == 1 && cols[j] == 1)) continue;
                count++;
            }
        }

        return count;
    }
}